File: library/src/main/java/cn/hx/prioritydialog/ActivityDialogHost.java
package cn.hx.prioritydialog;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentActivity;

public interface ActivityDialogHost extends DialogHost {

    void initAsDialogHost(@NonNull FragmentActivity activity);

    boolean warpStartActivityForResult(@NonNull Intent intent, int requestCode, @Nullable Bundle options);

    boolean warpFinish();
}

File: library/src/main/java/cn/hx/prioritydialog/ActivityDialogHost.kt
package cn.hx.prioritydialog

import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.FragmentActivity

interface ActivityDialogHost : DialogHost {
    fun initDialogHost(activity: FragmentActivity)

    fun warpStartActivityForResult(intent: Intent, requestCode: Int, options: Bundle?): Boolean

    fun warpFinish(): Boolean
}